RetailCoder.VBE/Refactorings/Rename/RenameModel.cs

```diff
@@ -76,5 +108,41 @@ public void PromptIfTargetImplementsInterface(ref Declaration target)
 
             target = interfaceMember;
         }
+
+        internal class Rewriters
+        {
+            Dictionary<string, IModuleRewriter> _rewriters;
+            RubberduckParserState _state;
+
+            public Rewriters(RubberduckParserState state)
+            {
+                _rewriters = new Dictionary<string, IModuleRewriter>();
+                _state = state;
+            }
+
+            public IModuleRewriter GetRewriter(QualifiedModuleName qmn)
+            {
+                IModuleRewriter rewriter;
+                if (_rewriters.ContainsKey(qmn.Name))
+                {
+                    _rewriters.TryGetValue(qmn.Name, out rewriter);
```

Comment:
@BZngr `_rewriters` should end up completely removed anyway. Get the rewriter from the parser state whenever you need it - don't cache it, it's *already* cached in the parser state.